### PR TITLE
IJPS-7a

### PR DESCRIPTION
**Pull request by raakheerele18**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/IJPS-7

### Change description ###
UI: Implement Login screen

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,6 +10,20 @@
 [![Maintainability Rating](https://sonarcloud.io/api/project_badges/measure?project=jps-judicial-payment-frontend&metric=sqale_rating)](https://sonarcloud.io/summary/new_code?id=jps-judicial-payment-frontend) 
 [![Coverage](https://sonarcloud.io/api/project_badges/measure?project=jps-judicial-payment-frontend&metric=coverage)](https://sonarcloud.io/summary/new_code?id=jps-judicial-payment-frontend)
 
+
+## Startup the Node service locally
+
+1. Make sure you have local-development.json within /config, if you do not you can get this from an XUI team member.
```
Comment (RebeccaBaker) [inline README.md:16]:
```suggestion
1. Make sure you have local-development.json within /config, if you do not you can get this from an JPS team member.
```

File: api/auth/index.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    callbackURL: getConfigValue(SERVICES_IDAM_OAUTH_CALLBACK_URL),
+    clientID: idamClient,
+    clientSecret: secret,
+    discoveryEndpoint: `${idamWebUrl}/o/.well-known/openid-configuration`,
+    issuerURL: issuerUrl,
+    logoutURL: idamApiPath,
+    responseTypes: ['code'],
+    routeCredential,
+    scope: 'profile openid roles manage-user create-user search-user',
+    sessionKey: 'xui-webapp',
```
Comment (RebeccaBaker) [inline api/auth/index.ts:102]: we need to use a new session key

File: api/auth/index.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    useRoutes: true,
+  };
+
+  const baseStoreOptions = {
+    cookie: {
+      httpOnly: true,
+      maxAge: 28800000,
+      secure: showFeature(FEATURE_SECURE_COOKIE_ENABLED),
+    },
+    name: 'xui-webapp',
```
Comment (RebeccaBaker) [inline api/auth/index.ts:114]: can this be jps and not xui please

File: config/default.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    "environment": "production",
+    "cookies": {
+        "token": "__auth__",
+        "userId": "__userid__"
+    },
+    "now": false,
+    "services": {
+        "idam": {
+            "idamApiUrl": "https://idam-api.platform.hmcts.net",
+            "idamClientID": "xuiwebapp",
```
Comment (RebeccaBaker) [inline config/default.json:11]: cant use xuiwebapp as a client id as that is for ExUi

File: config/default.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+            "idamLoginUrl": "https://hmcts-access.service.gov.uk",
+            "oauthCallbackUrl": "oauth2/callback",
+            "iss": "https://forgerock-am.service.core-compute-idam-prod.internal:8443/openam/oauth2/hmcts"
+          },
+          "role_assignment": {
+            "roleApi": "http://am-role-assignment-service-prod.service.core-compute-prod.internal"
+          },
+        "s2s": "http://rpe-service-auth-provider-prod.service.core-compute-prod.internal"
+    },
+    "microservice": "xui_webapp",
```
Comment (RebeccaBaker) [inline config/default.json:21]: our microservice is JPS not XUI

File: README.md (hunk 1 above)
Comment (raakheerele18) [reply README.md:16]: Changed

File: api/auth/index.ts (hunk 3 above)
Comment (raakheerele18) [reply api/auth/index.ts:114]: Changed

File: config/default.json (hunk 5 above)
Comment (raakheerele18) [reply config/default.json:21]: Changed

File: config/default.json (hunk 4 above)
Comment (raakheerele18) [reply config/default.json:11]: Changed

File: config/default.json [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        "idam": {
+            "idamApiUrl": "https://idam-api.platform.hmcts.net",
+            "idamClientID": "jpswebapp",
+            "idamLoginUrl": "https://hmcts-access.service.gov.uk",
+            "oauthCallbackUrl": "oauth2/callback",
+            "iss": "https://forgerock-am.service.core-compute-idam-prod.internal:8443/openam/oauth2/hmcts"
+          },
+          "role_assignment": {
+            "roleApi": "http://am-role-assignment-service-prod.service.core-compute-prod.internal"
+          },
```
Comment (RebeccaBaker) [inline config/default.json:18]: unused

File: config/custom-environment-variables.json [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        "idam": {
+            "idamApiUrl": "SERVICES_IDAM_API_URL",
+            "idamClientID": "SERVICES_IDAM_CLIENT_ID",
+            "idamLoginUrl": "SERVICES_IDAM_LOGIN_URL",
+            "oauthCallbackUrl": "SERVICES_IDAM_OAUTH_CALLBACK_URL",
+            "iss": "SERVICES_IDAM_ISS_URL"
+        },
+        "role_assignment": {
+            "roleApi": "SERVICES_ROLE_ASSIGNMENT_API"
+        },
```
Comment (RebeccaBaker) [inline config/custom-environment-variables.json:21]: unused

File: api/configuration/references.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+ */
+export const S2S_SECRET = 'secrets.rpx.mc-s2s-client-secret';
+export const REDIS_CLOUD_URL = 'secrets.rpx.webapp-redis-connection-string';
+
+export const COOKIES_TOKEN = 'cookies.token';
+export const COOKIES_USER_ID = 'cookies.userId';
+
+export const LOGGING = 'logging';
+
+export const SERVICES_ROLE_ASSIGNMENT_API_PATH = 'services.role_assignment.roleApi';
```
[the remark this answers is not in the document]
Comment (raakheerele18) [reply api/configuration/references.ts:19]: Removed

File: config/default.json (hunk 6 above)
Comment (raakheerele18) [reply config/default.json:18]: Removed

File: config/custom-environment-variables.json (hunk 7 above)
Comment (raakheerele18) [reply config/custom-environment-variables.json:21]: Removed

**Pull request merged**